virtual/environment.go

```diff
@@ -847,18 +891,36 @@ func (r *environment) isClosed() bool {
 	return r.shutdownState.closed
 }
 
+// pickServerForInvocation selects a server for invocation based on the provided references and retry policy.
+// It returns the selected actor reference, its index in the references slice, and an error if no references are available.
+// Note: The function may change the order of items in the references slice for efficiency and avoiding unnecessary copies.
 func (r *environment) pickServerForInvocation(
 	references []types.ActorReference,
 	create types.CreateIfNotExist,
-) (types.ActorReference, error) {
-	// TODO: implement invokation strategies in 'create' e.g. region-based, multi-invoke...
+) (types.ActorReference, int, error) {
 	if len(references) == 0 {
-		return types.ActorReference{}, fmt.Errorf("no references available")
+		return types.ActorReference{}, 0, fmt.Errorf("no references available")
+	}
+
+	switch create.Options.ReplicationStrategy {
+	case types.ReplicaSelectionStrategySorted:
+		// Sort the references slice based on the server ID in descending order.
+		// This way, the retry selection is biased towards the replica with the highest server ID over time.
+		sort.Slice(references, func(i, j int) bool {
```

Comment:
This scares me a bit because i'm not confident you're operating on a copy. Can you write a helper that just finds the min or max given a comparator instead?